### PR TITLE
keyvault: make commands work in cloud shell or vms with identity

### LINKED ISSUE #6313: az keyvault secret fails to get secret information for managed service identity
When you run this script on a VM with a managed service identity which has been granted full access to a key vault it returns nothing (not even an error):
```sh
az login --identity
az keyvault secret list --vault-name <vault-name>
az keyvault secret show --vault-name <vault-name> --name <secret-name>
```

Here is the output from --debug:
```
Command arguments: ['keyvault', 'secret', 'show', '--vault-name', '<vault-name>', '--name', '<secret-name>', '--debug']
Event: Cli.PreExecute []
Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x7f6b2839ebf8>, <function OutputProducer.on_global_arguments at 0x7f6b27ec68c8>, <function CLIQuery.on_global_arguments at 0x7f6b27a61400>]
Event: CommandInvoker.OnPreCommandTableCreate [<function CLILogging.remove_logger_flags at 0x7f6b2839ec80>]
Installed command modules ['acr', 'acs', 'advisor', 'ams', 'appservice', 'backup', 'batch', 'batchai', 'billing', 'cdn', 'cloud', 'cognitiveservices', 'configure', 'consumption', 'container', 'cosmosdb', 'dla', 'dls', 'dms', 'eventgrid', 'eventhubs', 'extension', 'feedback', 'find', 'interactive', 'iot', 'keyvault', 'lab', 'monitor', 'network', 'profile', 'rdbms', 'redis', 'reservations', 'resource', 'role', 'servicebus', 'servicefabric', 'sql', 'storage', 'vm']
Loaded module 'acr' in 0.002 seconds.
Loaded module 'acs' in 0.003 seconds.
Loaded module 'advisor' in 0.001 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'ams' in 0.003 seconds.
Loaded module 'appservice' in 0.004 seconds.
Loaded module 'backup' in 0.002 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'batch' in 0.008 seconds.
Loaded module 'batchai' in 0.003 seconds.
Loaded module 'billing' in 0.001 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'cdn' in 0.002 seconds.
Loaded module 'cloud' in 0.002 seconds.
Loaded module 'cognitiveservices' in 0.001 seconds.
Loaded module 'configure' in 0.000 seconds.
Loaded module 'consumption' in 0.004 seconds.
Loaded module 'container' in 0.001 seconds.
Loaded module 'cosmosdb' in 0.003 seconds.
Loaded module 'dla' in 0.003 seconds.
Loaded module 'dls' in 0.002 seconds.
Loaded module 'dms' in 0.002 seconds.
Loaded module 'eventgrid' in 0.002 seconds.
Loaded module 'eventhubs' in 0.003 seconds.
Loaded module 'extension' in 0.001 seconds.
Loaded module 'feedback' in 0.001 seconds.
Loaded module 'find' in 0.001 seconds.
Loaded module 'interactive' in 0.000 seconds.
Loaded module 'iot' in 0.003 seconds.
Loaded module 'keyvault' in 0.004 seconds.
Loaded module 'lab' in 0.002 seconds.
Loaded module 'monitor' in 0.004 seconds.
Loaded module 'network' in 0.018 seconds.
Loaded module 'profile' in 0.002 seconds.
Loaded module 'rdbms' in 0.003 seconds.
Loaded module 'redis' in 0.002 seconds.
Loaded module 'reservations' in 0.001 seconds.
Loaded module 'resource' in 0.004 seconds.
Loaded module 'role' in 0.002 seconds.
Loaded module 'servicebus' in 0.004 seconds.
Loaded module 'servicefabric' in 0.001 seconds.
Loaded module 'sql' in 0.004 seconds.
Event: CommandLoader.OnLoadCommandTable []
Loaded module 'storage' in 0.018 seconds.
Loaded module 'vm' in 0.009 seconds.
Loaded all modules in 0.137 seconds. (note: there's always an overhead with the first module loaded)
Extensions directory: '/root/.azure/cliextensions'
Event: CommandInvoker.OnPreCommandTableTruncate []
keyring.backend : Loading SecretService
keyring.backend : Loading kwallet
keyring.backend : Loading macOS
keyring.backend : Loading windows
Event: CommandInvoker.OnPostCommandTableCreate [<function add_id_parameters at 0x7f6b25c25048>]
Event: CommandInvoker.OnCommandTableLoaded []
Event: CommandInvoker.OnPreParseArgs [<function _documentdb_deprecate at 0x7f6b25a4c268>]
Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x7f6b27ec6950>, <function CLIQuery.handle_query_parameter at 0x7f6b27a61488>, <function handler at 0x7f6b2573e488>]
msrest.pipeline : Configuring request: timeout=100, verify=True, cert=None
msrest.pipeline : Configuring proxies: ''
msrest.pipeline : Evaluate proxies against ENV settings: True
msrest.pipeline : Configuring redirects: allow=True, max=30
msrest.pipeline : Configuring retry: max_retries=4, backoff_factor=0.8, max_backoff=90
urllib3.connectionpool : Starting new HTTPS connection (1): <vault-name>.vault.azure.net
urllib3.connectionpool : https://<vault-name>.vault.azure.net:443 "GET /secrets/<secret-name>/?api-version=2016-10-01 HTTP/1.1" 401 0
Event: CommandInvoker.OnTransformResult [<function _resource_group_transform at 0x7f6b25c3f6a8>, <function _x509_from_base64_to_hex_transform at 0x7f6b25c3f730>]
Event: CommandInvoker.OnFilterResult []
Event: Cli.PostExecute []
```

Manually retrieving a bearer token and reading the same secret works:
```
curl 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fvault.azure.net' -H Metadata:true
curl 'https://<vaultname>.vault.azure.net/secrets/<secret-name>?api-version=2016-10-01' -H "Authorization: Bearer <auth-token>"
```

### PR DESCRIPTION
**Pull request by yugangw-msft**
Fix #6313 

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/6319
This is an **experimental preview** for @microsoft users.

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_command_type.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def keyvault_exception_handler(ex):
                            'The vault may not exist or you may need to flush your DNS cache '
                            'and try again later.')
         raise CLIError(ex)
+    else:
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_command_type.py]: @tjprescott, this is an optional change, if you feel it has side impact, I can revert/improve. So far this error handler swallows exceptions and hides the real issue.

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_client_factory.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ def get_token(server, resource, scope):  # pylint: disable=unused-argument
         import adal
         from azure.cli.core._profile import Profile
         try:
-            return Profile(cli_ctx=cli_ctx).get_login_credentials(resource)[0]._token_retriever()  # pylint: disable=protected-access
+            return Profile(cli_ctx=cli_ctx).get_raw_token(resource)[0]
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_client_factory.py]: I didn't write tests in this PR as i feel the existing testbed should be sufficient to capture potential regressions

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_command_type.py [hunk 3]
```diff
@@ -43,8 +43,6 @@ def keyvault_exception_handler(ex):
                            'The vault may not exist or you may need to flush your DNS cache '
                            'and try again later.')
         raise CLIError(ex)
-    else:
-        raise ex
```
Comment (tjprescott) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_command_type.py]: What fails with this?

Comment (yugangw-msft) [reply src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_command_type.py]: See below. I can still throw non CLIError, but need to confirmation/encourage from your guys to proceed

```Python
    @ResourceGroupPreparer(name_prefix='cli_test_kv_cert_issuer')
    def test_keyvault_certificate_issuers(self, resource_group):
       .....
        # test admin commands
        self.cmd('keyvault certificate issuer admin add --vault-name {kv} --issuer-name issuer1 --email test@test.com --first-name Test --last-name Admin --phone 123-456-7890', checks=[
            ...
        ])
        # ---> fail below which i guess caused by adding dupe issuers<-----
        self.cmd('keyvault certificate issuer admin add --vault-name {kv} --issuer-name issuer1 --email test@test.com')  
```
```
  File "d:\sdk\azure-cli\src\command_modules\azure-cli-keyvault\azure\cli\command_modules\keyvault\_command_type.py", line 118, in keyvault_command_handler
    return keyvault_exception_handler(ex)
  File "d:\sdk\azure-cli\src\command_modules\azure-cli-keyvault\azure\cli\command_modules\keyvault\_command_type.py", line 47, in keyvault_exception_handler
    raise ex
  File "d:\sdk\azure-cli\src\command_modules\azure-cli-keyvault\azure\cli\command_modules\keyvault\_command_type.py", line 98, in keyvault_command_handler
    result = op(**command_args)
  File "d:\sdk\azure-cli\src\command_modules\azure-cli-keyvault\azure\cli\command_modules\keyvault\custom.py", line 800, in add_certificate_issuer_admin
    raise CLIError("admin '{}' already exists".format(email))
knack.util.CLIError: admin 'test@test.com' already exists
```

**Pull request merged**